context.go

```diff
@@ -21,25 +21,61 @@
 package fx
 
 import (
-	gcontext "context"
+	"context"
 
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber/jaeger-client-go"
+	"go.uber.org/fx/service"
 	"go.uber.org/fx/ulog"
 )
 
-// Context embeds Host and Go stdlib context for use
-type Context interface {
-	gcontext.Context
+type contextKey int
 
-	Logger() ulog.Log
+const _contextStore contextKey = iota
+
+type ctxStore struct {
```

Comment:
Why do you prefer to have a struct over storing e.g. ulog.Log directly in the context?
Each module can provide functions to work with context, which gives nice separation..

The only bad part I see is performance: access to elements in context seems to be linear:
https://golang.org/src/context/context.go?s=7222:7247#L469
But it shouldn't be a big deal, we'll have only few elements.